urla39b63eb76_tef_pipeline_tgz-pJ8-org_twentygototen_pipeline_ExampleJ8/src/org/twentygototen/pipeline/pipes/FileReaderPipe.java

```diff
@@ -0,0 +1,31 @@
+/**
+ * 
+ */
+package org.twentygototen.pipeline.pipes;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileReader;
+import java.util.ArrayList;
+import java.util.List;
+
+import org.twentygototen.pipeline.util.Pair;
+
+final public class FileReaderPipe extends SyncPipe<File, Pair<File, List<String>>> {
+	@Override
+	final public Pair<File,List<String>> process(File input) {
+		List<String> out =  new ArrayList<String>();
+		try {
+			BufferedReader bufferedReader = new BufferedReader(new FileReader(input));
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code might contain a resource leak. Resource leaks can cause your system to slow down or crash.

**Fix**
Consider closing the following resource: *bufferedReader*. Execution paths that do not contain closure statements were detected. Ensure *bufferedReader* is closed in a try-finally block or declared in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).